### ISSUE TITLE
(lambda-nodejs): Maximum call stack size exceeded when providing a Vpc

### ISSUE THREAD

**Issue by guillaumemaka** [aws/aws-cdk#11862]
When passing a Vpc to NodeJsFunction throw Maximum call stack size exceeded

### Reproduction Steps

```
// const vpc = Vpc.fromLookup(this, _id('BaseInfraVpc'), {
//   vpcId: '....'
// });
// or
const vpc = new Vpc(this, 'vpc', {
   maxAzs: 1,
});

const fn = new NodejsFunction(this, _id('LambdaSendEmail'), {
  depsLockFilePath: path.resolve(__dirname, '..', '..', 'yarn.lock'),
  minify: false,
  target: 'node12.19.0',
  entry: path.resolve(__dirname, '..', '..', 'src', 'index.ts'),
  handler: 'handler',
  environment: {
    ....
  },
  vpc: vpc,
  timeout: cdk.Duration.seconds(30),
});

```
### What did you expect to happen?

Lambda function build with no error

### What actually happened?

```
CDK toolkit version: 1.76.0 (build c207717)
Command line arguments: {
  _: [ 'diff' ],
  profile: '<redacted>',
  verbose: 1,
  v: 1,
  debug: true,
  lookups: true,
  'ignore-errors': false,
  ignoreErrors: false,
  json: false,
  j: false,
  ec2creds: undefined,
  i: undefined,
  'version-reporting': undefined,
  versionReporting: undefined,
  'path-metadata': true,
  pathMetadata: true,
  'asset-metadata': true,
  assetMetadata: true,
  'role-arn': undefined,
  r: undefined,
  roleArn: undefined,
  staging: true,
  'no-color': false,
  noColor: false,
  fail: false,
  'context-lines': 3,
  contextLines: 3,
  strict: false,
  '$0': 'cdk'
}
cdk.json: {
  "app": "npx ts-node --prefer-ts-exts src/main.ts"
}
cdk.context.json: {
  "ssm:account=<redacted>:parameterName=<redacted>:region=ca-central-1": "<redacted>"
}
merged settings: {
  versionReporting: true,
  pathMetadata: true,
  output: 'cdk.out',
  app: 'npx ts-node --prefer-ts-exts src/main.ts',
  context: {},
  debug: true,
  assetMetadata: true,
  profile: '<redacted>',
  toolkitBucket: {},
  staging: true,
  bundlingStacks: [ '*' ],
  lookups: true
}
Toolkit stack: CDKToolkit
Setting "CDK_DEFAULT_REGION" environment variable to us-east-1
Resolving default credentials
Retrieved account ID <redacted> from disk cache
Setting "CDK_DEFAULT_ACCOUNT" environment variable to <redacted>
context: {
  'ssm:account=<redacted>:parameterName=/habilhome/BaseVpcId:region=ca-central-1': '<redacted>',
  'aws:cdk:enable-path-metadata': true,
  'aws:cdk:enable-asset-metadata': true,
  'aws:cdk:version-reporting': true,
  'aws:cdk:bundling-stacks': [ '*' ]
}
outdir: cdk.out
env: {
  CDK_DEFAULT_REGION: 'us-east-1',
  CDK_DEFAULT_ACCOUNT: '<redacted>',
  CDK_DEBUG: 'true',
  CDK_CONTEXT_JSON: '{"ssm:account=<redacted>:parameterName=<redacted>:region=ca-central-1":"<redacted>","aws:cdk:enable-path-metadata":true,"aws:cdk:enable-asset-metadata":true,"aws:cdk:version-reporting":true,"aws:cdk:bundling-stacks":["*"]}',
  CDK_OUTDIR: 'cdk.out',
  CDK_CLI_ASM_VERSION: '7.0.0',
  CDK_CLI_VERSION: '1.76.0'
}
Maximum call stack size exceeded
Subprocess exited with error 1
Error: Subprocess exited with error 1
    at ChildProcess.<anonymous> (/Users/guillaume/.nvm/versions/node/v14.15.0/lib/node_modules/aws-cdk/lib/api/cxapp/exec.ts:122:23)
    at ChildProcess.emit (events.js:315:20)
    at ChildProcess.EventEmitter.emit (domain.js:486:12)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:277:12)
```

### Environment

  - **CDK CLI Version  : 1.76.0
  - **Framework Version:**
  - **Node.js Version: v14.15.0, v12.19.1
  - **OS               : macOS Catalina 10.15.7 (19H15)
  - **Language (Version): ts-node@^9.0.0, typescript@^3.9.5

--- 

This is :bug: Bug Report

**Comment by jogold** [comment]
This is #11693, fixed (#11698)  but not released yet.

Temporary workaround https://github.com/aws/aws-cdk/issues/11693#issuecomment-737262463

**Comment by jogold** [comment]
@eladb this can be closed now.
